src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests.cs

```diff
@@ -45,5 +45,89 @@ public static void Main(string[] args)
+                // (6,40): error CS8059: Feature 'throw expression' is not available in C# 6.  Please use language version 7 or greater.
+                //         void NeverReturnsFunction() => throw new NullReferenceException();
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion6, "throw new NullReferenceException()").WithArguments("throw expression", "7").WithLocation(6, 40),
+                // (7,21): error CS8059: Feature 'throw expression' is not available in C# 6.  Please use language version 7 or greater.
+                //         int x = b ? throw new NullReferenceException() : 1;
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion6, "throw new NullReferenceException()").WithArguments("throw expression", "7").WithLocation(7, 21),
+                // (8,21): error CS8059: Feature 'throw expression' is not available in C# 6.  Please use language version 7 or greater.
+                //         x = b ? 2 : throw new NullReferenceException();
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion6, "throw new NullReferenceException()").WithArguments("throw expression", "7").WithLocation(8, 21),
+                // (9,18): error CS8059: Feature 'throw expression' is not available in C# 6.  Please use language version 7 or greater.
+                //         s = s ?? throw new NullReferenceException();
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion6, "throw new NullReferenceException()").WithArguments("throw expression", "7").WithLocation(9, 18),
+                // (11,47): error CS8059: Feature 'throw expression' is not available in C# 6.  Please use language version 7 or greater.
+                //         throw new NullReferenceException() ?? throw new NullReferenceException() ?? throw null;
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion6, "throw new NullReferenceException() ?? throw null").WithArguments("throw expression", "7").WithLocation(11, 47),
+                // (11,85): error CS8059: Feature 'throw expression' is not available in C# 6.  Please use language version 7 or greater.
+                //         throw new NullReferenceException() ?? throw new NullReferenceException() ?? throw null;
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion6, "throw null").WithArguments("throw expression", "7").WithLocation(11, 85),
+                // (13,42): error CS8059: Feature 'throw expression' is not available in C# 6.  Please use language version 7 or greater.
+                //     public static void NeverReturns() => throw new NullReferenceException();
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion6, "throw new NullReferenceException()").WithArguments("throw expression", "7").WithLocation(13, 42)
+                );
+        }
+
+        [Fact]
+        public void ThrowExpression_Bad()
```

Comment:
The list could get long here, since most contexts disallowed, but consider throw in tuple and `return throw ...`.

Also, this makes me wonder about refactoring scenarios. Let's say that you have an expression-bodied method that throws, is there a refactoring that inlines the method? 
Or are there refactorings on conditional or coalescing expressions that might push the `throw` to a different (forbidden) parent?